syntax = "proto3";

package rtabmap.proto;

import "Transform.proto";

message CameraParameters
{
    int32 width = 1;
    int32 height = 2;
    float fx = 3;
    float fy = 4;
    float cx = 5;
    float cy = 6;
}

message CameraImage
{
    int32 rows = 1;
    int32 cols = 2;
    int32 type = 3;
    bytes data = 4;
}

message CameraData
{
    Transform to_sensor = 1;
    CameraParameters parameters = 2;
    CameraImage image = 3;
}

message LidarData
{
    Transform to_sensor = 1;
    repeated float points = 2;  // xyz
}

message SensorData
{
    repeated CameraData cameras_data = 1;
    repeated LidarData lidars_data = 2;
}
